corehq/apps/hqwebapp/templatetags/hq_shared_tags.py

```diff
@@ -332,3 +332,32 @@ def render(self, context):
         if nodelist is None:
             return ""
         return nodelist.render(context)
+
+
+# https://djangosnippets.org/snippets/545/
+@register.tag(name='captureas')
+def do_captureas(parser, token):
+    """
+    Assign to a context variable from within a template
+        {% capturas my_context_var %}<!-- anything -->{% endcaptureas %}
+        <h1>Nice job capturing {{ my_context_var }}</h1>
+    """
+    try:
+        tag_name, args = token.contents.split(None, 1)
+    except ValueError:
+        raise template.TemplateSyntaxError("'captureas' node requires a "
+                                           "variable name.")
+    nodelist = parser.parse(('endcaptureas',))
+    parser.delete_first_token()
+    return CaptureasNode(nodelist, args)
+
+
+class CaptureasNode(template.Node):
+    def __init__(self, nodelist, varname):
+        self.nodelist = nodelist
+        self.varname = varname
+
+    def render(self, context):
+        output = self.nodelist.render(context)
+        context[self.varname] = output
+        return ''
```

Comment:
This whole thing is copypasta'd from https://djangosnippets.org/snippets/545/ except for the docstring.  I googled this problem a bit and found people recommending that.